src/wstool/multiproject_cli.py

```diff
@@ -621,6 +622,99 @@ def cmd_diff(self, target_path, argv, config=None):
 
         return False
 
+    def cmd_foreach(self, target_path, argv, config=None):
+        """Run shell commands in each repository."""
+        parser = OptionParser(
+            usage='usage: %s foreach [localname]* [command] [OPTIONS]' % self.progname,
```

Comment:
I believe combining localnames and scmtypes is confusing. If the user runs ```wstool foreach foo 'ls' --git```, and foo is not of type git, then the tool could either do this for all git repos and foo, or for no repo. The user might expect either behavior, and be surprised if the behavior is the opposite.

If you think we need either possibility, I would prefer them to be distinct, so usage could be:
```
usage: wstool foreach [[localname]* | [VCSFILTER]*] [command] [OPTIONS]
```
It does not matter too much here whether VCSFILTER options could also be placed after command. The tool should merely throw an error if users try to provide both, because the tool cannot guess the users intention.